week-2-wide-perspective/w02e01-functional-warm-up/src/helpers.js

```diff
@@ -0,0 +1,61 @@
+export const getLength = (/** @type {object[]} */ arr) => {
+  if (!Array.isArray(arr)) {
+    throw new Error('arr must be of the array type');
+  }
+  return arr.length;
+};
+
+export const getByProperty = (/** @type {object} */ obj, /** @type {string[]} */ ...properties) => {
+  let tempObj = obj;
+  if (!tempObj) {
+    throw new Error('Object is required');
+  }
+  if (!properties) {
+    throw new Error('Property is required');
+  }
+  properties.forEach((property) => {
+    if (!tempObj[property] && tempObj[property] !== 0) {
```

Comment:
masz coś takiego jak `in`

np:
```javascript
if(!(property in tempObj)) {
 .... throw
}
```
Będzie to czytelniejsze tutaj.

Reply:
tylko trzeba uważać na `undefined` (choć to mało prawdopodobne żeby ktoś na obiekcie zdefiniował `undefined` ale no może się tak zdarzyć )

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in